pkg/sources/transformer/builtin/event_time/event_time_extractor.go

```diff
@@ -80,8 +80,8 @@ func (e eventTimeExtractor) apply(et time.Time, payload []byte) (functionsdk.Mes
 		newEventTime, err = dateparse.ParseStrict(timeStr)
 	}
 	if err != nil {
-		return functionsdk.MessageTToAll(et, payload), err
+		return functionsdk.NewMessageT(payload).WithEventTime(et), err
```

Comment:
the use case of the transformer is not just to change the event time of the message right? if the udf developer doesn't set the event time (that means they don't care about the event time, they could be using the transformer for other purposes) for such cases we should use the parent message's event time, right? My understanding is transformers are used to transform the input message in the source vertex itself to avoid one more map vertex, it's not just limited to setting new event time.

Reply:
Discussed offline, will use public constructors to enforce specifying event time.